src/DataAggregator.cpp

```diff
@@ -264,22 +269,46 @@ void DataAggregator::launchPerfProcess(StringRef Name, PerfProcessInfo &PPI,
   Argv.push_back(Filename.c_str());
   Argv.push_back(nullptr);
 
-  if (auto Errc = sys::fs::createTemporaryFile("perf.script", "out",
-                                               PPI.StdoutPath)) {
-    errs() << "PERF2BOLT: failed to create temporary file "
-           << PPI.StdoutPath << " with error " << Errc.message()
-           << "\n";
-    exit(1);
-  }
-  TempFiles.push_back(PPI.StdoutPath.data());
+  if(opts::TmpDirectory.empty()) {
```

Comment:
Ideally, use clang format (only on the lines that you are changing) for correct indentation.
`if (opts::TmpDirectory.empty())`

Reply:
Oops that was a typo.